docs/content/latest/api/ysql/datatypes/type_array/functions-operators/comparison.md

```diff
@@ -0,0 +1,453 @@
+
+```postgresql
+with
+  v as (
+    select
+      (select array['a', 'b', 'c',  'd']::text[]) as a1,
+      (select array['a',      'c'      ]::text[]) as a2
+  )
+select
+  (a1 @> a2)::text as "CONTAINS comparison result",
+  (a2 <@ a1)::text as "'IS CONTAINED BY' comparison result"
+from v;
+```
+This is the result:
+```
+ CONTAINS comparison result | 'IS CONTAINED BY' comparison result 
+----------------------------+-------------------------------------
+ true                       | true
+```
+
+### The `&&` operator
+
+The `&&` operator returns `true` if the LHS and RHS arrays overlap—i.e. if they have at least one value in common. The definition of this operator makes it insensitive to which of the two to-be-compared is used on the LHS and which is used on the RHS.
+
+```postgresql
+with
+  v as (
+    select
+      (select array['a', 'b', 'c',  'd']::text[]) as a1,
+      (select array['d', 'e', 'f',  'g']::text[]) as a2
+  )
+select
+  (a1 && a2)::text as "'a1 OVERLAPS a2' comparison result",
+  (a2 && a1)::text as "'a2 OVERLAPS a1' comparison result"
+from v;
+```
+This is the result:
+```
+ 'a1 OVERLAPS a2' comparison result | 'a2 OVERLAPS a1' comparison result 
+------------------------------------+------------------------------------
+ true                               | true
+```
+
+## Equality and inequality semantics
+
+This section demonstrates each of the rules that the _"Comparison operators overview"_ section, [above](./#comparison-operators-overview), stated.
+
+```postgresql
+-- Any two arrays can be compared without error if they have the same datatype.
+do $body$
+begin
+  ------------------------------------------------------------------------------
+  -- Illustrate "IS NOT DISTINCT FROM" semantics.
+  declare
+    v1 constant int := 1;
+    v2 constant int := 1;
+    n1 constant int := null;
+    n2 constant int := null;
+  begin
+    assert
+      (v1 = v2)                    and
+      (v1 is not distinct from v2) and
+
+      ((n1 = n2) is null)          and
+      (n1 is not distinct from n2),
+    'unexpected';
+  end;
+
+  ------------------------------------------------------------------------------
+  -- Basic demonstration of equaliy when the geom. properties of
+  -- the two arrays are identical.
+  -- Shows that pairwise comparison uses ""IS NOT DISTINCT FROM" semantics and NOT
+  -- the conventional NULL semantics used when scalars are compared.
+  declare
+    a constant int[] := '{10, null, 30}';
+    b constant int[] := '{10, null, 30}'; -- Identical to a.
+  begin
+    assert
+      (a = b),
+    '"a = b" assert failed';
+
+    -- Because of this, we don't need ever to write this.
+    assert
+      (a is not distinct from b),
+    '"a is not distinct from b" assert failed';
+  end;
+
+  ------------------------------------------------------------------------------
+  -- Basic demonstration of inequaliy when the geom. properties of
```

Comment:
```suggestion
  -- Basic demonstration of inequality when the geometric properties of
```

Reply:
Of course code comments should ne fee of spelling errors. Do you have a spell checker that handles these sensibly without annoying false positives from the code itself?

But are abbreviations in comments really disallowed. Anything that readuces line length and the line count is highly desirable in code illustrations.